core/string/print_string.cpp

```diff
@@ -74,6 +74,52 @@ void __print_line(String p_string) {
 		return;
 	}
 
+	// Removing all bbcode from normal print string:
+	p_string = p_string.replace("[b]", "");
+	p_string = p_string.replace("[/b]", "");
+	p_string = p_string.replace("[i]", "");
+	p_string = p_string.replace("[/i]", "");
+	p_string = p_string.replace("[u]", "");
+	p_string = p_string.replace("[/u]", "");
+	p_string = p_string.replace("[s]", "");
+	p_string = p_string.replace("[/s]", "");
+	p_string = p_string.replace("[indent]", "");
+	p_string = p_string.replace("[/indent]", "");
+	p_string = p_string.replace("[code]", "");
+	p_string = p_string.replace("[/code]", "");
+	p_string = p_string.replace("[center]", "");
+	p_string = p_string.replace("[/center]", "");
+	p_string = p_string.replace("[right]", "");
+	p_string = p_string.replace("[/right]", "");
+	p_string = p_string.replace("[color=black]", "");
+	p_string = p_string.replace("[color=red]", "");
+	p_string = p_string.replace("[color=green]", "");
+	p_string = p_string.replace("[color=lime]", "");
+	p_string = p_string.replace("[color=yellow]", "");
+	p_string = p_string.replace("[color=blue]", "");
+	p_string = p_string.replace("[color=magenta]", "");
+	p_string = p_string.replace("[color=pink]", "");
+	p_string = p_string.replace("[color=purple]", "");
+	p_string = p_string.replace("[color=cyan]", "");
+	p_string = p_string.replace("[color=white]", "");
+	p_string = p_string.replace("[color=orange]", "");
+	p_string = p_string.replace("[color=gray]", "");
+	p_string = p_string.replace("[/color]", "");
+	p_string = p_string.replace("[bgcolor=black]", "");
+	p_string = p_string.replace("[bgcolor=red]", "");
+	p_string = p_string.replace("[bgcolor=green]", "");
+	p_string = p_string.replace("[bgcolor=lime]", "");
+	p_string = p_string.replace("[bgcolor=yellow]", "");
+	p_string = p_string.replace("[bgcolor=blue]", "");
+	p_string = p_string.replace("[bgcolor=magenta]", "");
+	p_string = p_string.replace("[bgcolor=pink]", "");
+	p_string = p_string.replace("[bgcolor=purple]", "");
+	p_string = p_string.replace("[bgcolor=cyan]", "");
+	p_string = p_string.replace("[bgcolor=white]", "");
+	p_string = p_string.replace("[bgcolor=orange]", "");
+	p_string = p_string.replace("[bgcolor=gray]", "");
+	p_string = p_string.replace("[/bgcolor]", "");
+
```

Comment:
Normal printing shouldn't care about these. If I'd write `print("[b]")` I'd expect `[b]` to be printed etc.

```suggestion
```

Reply:
The problem with this is that else you can also use bbcode in normal print. How could we solve that problem then?

Reply:
@kleonc How about the update which I've done, it prints [b] and such in the terminal but still removes it for the editor console. I can't really find another way of doing this, else the normal print("[b]") also makes the text bold because the text gets appended. I tried figuring out how to make the text only append for print_rich(), but they all failed as I could not give them a new p_type like MSG_TYPE_RICH. All messages to the log get MSG_TYPE_STD no matter what I did.

So I hope this is okay that it prints the raw bbcode in the terminal but not in the editor console. I updated the picture in the first comment with an example of how it looks now.

Reply:
Oh, you mean that `print("[b]text[/b]")` would print bold `text` in the editor because in-editor console is a `RichTextLabel` with bbcode enabled? Hmm, that's unfortunate. I don't think `RichTextLabel` allows to ignore bbcode in some part of its assigned text (it either parses all or nothing?:thinking:). So to solve it properly probably `RichTextLabel` would need to be modified to somehow allow partially disabling bbcode parsing. Not sure. Alternatively multiple `RichTextLabel`s could be used (so only some of them would parse bbcode) but I don't think it would work well, it would probably cause problems with text selecting etc.
So I don't see a simple solution.

But again: if `print("[b]text[/b]")` will print just `text` then I think it's a problem.

Reply:
> @kleonc How about the update which I've done, it prints [b] and such in the terminal but still removes it for the editor console. I can't really find another way of doing this, else the normal print("[b]") also makes the text bold because the text gets appended. I tried figuring out how to make the text only append for print_rich(), but they all failed as I could not give them a new p_type like MSG_TYPE_RICH. All messages to the log get MSG_TYPE_STD no matter what I did.
> 
> So I hope this is okay that it prints the raw bbcode in the terminal but not in the editor console. I updated the picture in the first comment with an example of how it looks now.

Hmm, I think it might be acceptable/good enough if this will be well documented. Solving it properly seems not so straighforward and I'm not sure if it's really worth the effort.
Of course that's just my opinion, I'm not anyhow decisive here.

Reply:
Well I kind of find that it should be like how you say, when doing print("[b]") that it actually print [b].
It's not easy but I'm slowly getting to a solution on how to solve this. However the new console system in the editor is putting up a fight as I can't just create a new type of message. Not sure yet how to solve it, but I'm halfway there. ^^

Reply:
@kleonc I got good news! I updated the first comment if you want to see more of the information. But basically I was able to make it so print(...) prints everything in a normal way again like it does now. And only when using print_rich(...) you will be able to use BBCodes.

I hope you are happy with this change. I am most certainly happy as I can now say that I did a good job. This is my first contribution so I hope it gets accepted, but I'm already quite happy with what I achieved. Thanks for pointing this issue out, it motivated me to try one more time to improve this whole pull request. Thanks! 😄

Reply:
Nice, thanks! Seems like logging is way more customizable than I thought (yeah, I'm not familiar with the relevant code :laughing:).

Yes, I'm happy with this change behavior-wise. Can't really comment on the conceptual side of the code (as I'm not familiar with it and I didn't look into it deeply).